presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyCountOverConstant.java

```diff
@@ -105,11 +114,30 @@ private boolean isCountOverConstant(AggregationNode.Aggregation aggregation, Ass
             return false;
         }
 
-        Expression argument = aggregation.getArguments().get(0);
-        if (argument instanceof SymbolReference) {
-            argument = inputs.get(Symbol.from(argument));
+        RowExpression argument = aggregation.getArguments().get(0);
+        Expression assigned = null;
+        if (isExpression(argument)) {
+            if (castToExpression(argument) instanceof SymbolReference) {
+                assigned = inputs.get(Symbol.from(castToExpression(argument)));
+            }
+        }
+        else {
+            if (argument instanceof VariableReferenceExpression) {
+                assigned = inputs.get(toSymbol(argument));
+            }
+            else {
+                throw new UnsupportedOperationException();
+            }
         }
 
-        return argument instanceof Literal && !(argument instanceof NullLiteral);
+        return assigned instanceof Literal && !(assigned instanceof NullLiteral);
+    }
+    private static Symbol toSymbol(RowExpression rowExpression)
```

Comment:
This is the same to the one in `MultipleDistinctAggregationToMarkDistinct` and `SingleDistinctAggregationToGroupBy.java` right?